spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/config/DiscoveryClientConfigServiceAutoConfiguration.java

```diff
@@ -53,17 +52,18 @@
 	@Autowired
 	private EurekaDiscoveryClientConfiguration lifecycle;
 
+	@Autowired
+	private EurekaClient eurekaClient;
+
 	@PostConstruct
 	public void init() {
 		this.lifecycle.stop();
 		rebind(this.clientConfig, "eurekaClientConfig");
 		rebind(this.instanceConfig, "eurekaInstanceConfig");
-        if (DiscoveryManager.getInstance().getDiscoveryClient() != null) {
-            DiscoveryManager.getInstance().getDiscoveryClient().shutdown();
-        }
-		ApplicationInfoManager.getInstance().initComponent(this.instanceConfig);
-		DiscoveryManager.getInstance().initComponent(this.instanceConfig,
-				this.clientConfig);
+		eurekaClient.shutdown();;
+		// FIXME: reinit EurekaClient and ApplicationInfoManager
+		//applicationInfoManager.initComponent(this.instanceConfig);
+		//discoveryManager.initComponent(this.instanceConfig, this.clientConfig);
```

Comment:
@dsyer not sure what I need to do here and what case this is fixing to test it.  Now all the objects held in `ApplicationInfoManager` and `DiscoveryManager` are now beans.